### PR TITLE
solo2-cli 0.2.2 (new formula)

### PR DESCRIPTION
**Pull request by domrim**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

new version, based on #112877 and #107574

### CODE REVIEW

File: Formula/s/solo2-cli.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  def install
+    system "cargo", "install", "--all-features", *std_cargo_args
+
+    bash_completion.install "target/release/solo2.bash"
+    fish_completion.install "target/release/solo2.fish"
+    zsh_completion.install "target/release/_solo2"
+  end
+
+  test do
+    assert_equal "", shell_output("#{bin}/solo2 ls")
```
Comment (chenrui333) [inline Formula/s/solo2-cli.rb:32]:
```suggestion
    assert_empty shell_output("#{bin}/solo2 ls")
```

File: Formula/s/solo2-cli.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  url "https://github.com/solokeys/solo2-cli/archive/refs/tags/v0.2.2.tar.gz"
+  sha256 "49a30c5ee6f38be968a520089741f8b936099611e98e6bf2b25d05e5e9335fb4"
+  license any_of: ["Apache-2.0", "MIT"]
+  head "https://github.com/solokeys/solo2-cli.git", branch: "main"
+
+  livecheck do
+    url :stable
+    regex(/^?(\d+(?:\.\d+)+)$/i)
+  end
+
```
Comment (chenrui333) [inline Formula/s/solo2-cli.rb:13]:
```suggestion
```

**Review by chenrui333 — APPROVED** [review]
just two nit comments, LGTM

**Comment by domrim** [comment]
added your suggestions, thx ;)

**Review by chenrui333 — APPROVED** [review]
LGTM, thanks!

File: Formula/s/solo2-cli.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  url "https://github.com/solokeys/solo2-cli/archive/refs/tags/v0.2.2.tar.gz"
+  sha256 "49a30c5ee6f38be968a520089741f8b936099611e98e6bf2b25d05e5e9335fb4"
+  license any_of: ["Apache-2.0", "MIT"]
+  head "https://github.com/solokeys/solo2-cli.git", branch: "main"
+
+  depends_on "rust" => :build
+
+  on_linux do
+    depends_on "pkg-config" => :build
+    depends_on "libusb"
```
Comment (ZhongRuoyu) [inline Formula/s/solo2-cli.rb:13]: There is no linkage against `libusb`. Is it really used?

Comment (domrim) [reply Formula/s/solo2-cli.rb:13]: the hidapi crate either needs libusb or hidraw, see https://docs.rs/hidapi/latest/hidapi/#linux-backends

Comment (domrim) [reply Formula/s/solo2-cli.rb:13]: see also the comment in the AUR https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=solo2-cli

Comment (ZhongRuoyu) [reply Formula/s/solo2-cli.rb:13]: But it's statically linking against hidraw, which means libusb is not used: https://github.com/solokeys/solo2-cli/blob/v0.2.2/Cargo.toml#L26

```toml
hidapi = { version = "2", default-features = false, features = ["linux-static-hidraw"] }
```

Which should also be the reason why it has linkage against libudev, according to https://github.com/ruabmbua/hidapi-rs/blob/b69e5db0b7267901da65910d563e9d142e491948/build.rs#L58.

**Review by ZhongRuoyu — APPROVED** [review]
Thank you, @domrim.

**Pull request merged**